File: apps/python-api/src/yourmedportal_api/lib/email.py
from __future__ import annotations

import logging
from typing import Any, Literal

EmailKind = Literal[
    "booking_confirmed",
    "booking_cancelled",
    "report_created",
    "report_ready",
]

log = logging.getLogger(__name__)


def send_email_stub(
    kind: EmailKind,
    to: str | None,
    payload: dict[str, Any],
) -> None:
    if not to:
        return
    log.info("[email:%s] to=%s %s", kind, to, payload)

File: apps/api/src/lib/email.ts
type EmailKind =
	| 'booking_confirmed'
	| 'booking_cancelled'
	| 'report_created'
	| 'report_ready';

export function sendEmailStub(
	kind: EmailKind,
	to: string | null | undefined,
	payload: Record<string, unknown>
): void {
	if (!to) return;
	console.log(`[email:${kind}] to=${to}`, payload);
}
